onnxruntime/core/platform/posix/env.cc

```diff
@@ -278,10 +278,12 @@ class PosixEnv : public Env {
 
   // Return the number of physical cores
   int GetNumPhysicalCpuCores() const override {
+#if !defined(__APPLE__) && !defined(__ANDROID__) && !defined(__wasm__) && !defined(_AIX)
 #ifdef CPUINFO_SUPPORTED
```

Comment:
Can these checks be done in the cmake file when setting CPUINFO_SUPPORTED?

Reply:
I suspect not. Reason being, CPU info is used for processor capabilities checks and must be available on a wider set of platforms, than that of we use for affinity.

Reply:
Was more asking can the checks on platform be done in cmake when deciding what to set CPUINFO_SUPPORTED to so this code only need the CPUINFO_SUPPORTED ifdef?

Reply:
I think I understood you right. I don't think I can remove `CPUINFO_SUPPORTED` just because we do not do affinity on those platforms.